src/AbstractApplication.php

```diff
@@ -265,15 +273,21 @@ public function setResponse(ResponseInterface $response): ApplicationInterface
         
         return $this;
     }
-    
-    public function run()
+
+    /**
+     * @param EmitterInterface|null $emitter
+     * @return mixed|void
+     */
+    public function run(EmitterInterface $emitter = null)
```

Comment:
Wuuut ??? :-)

And what if constructor set a HTTP request ?

Reply:
It's a small change that let me mock the emitter to prevent "header already sent" error will running the test. And it also let users a way to change the Emitter.